### PR TITLE
Task/lattice-2238 — add filter option to select

### PR DESCRIPTION
**Pull request by giuliacampana**
make `Select` components more performative with 1000+ options by adding filter.

### CODE REVIEW

File: src/select/src/components/CheckboxSelect.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ export const defaultProps = {
   isMulti: true,
   hideSelectedOptions: false,
   closeMenuOnSelect: false,
+  filterOption: createFilter({ ignoreAccents: false }),
```
Comment (solomontang) [inline src/select/src/components/CheckboxSelect.js]: sort these props please.

File: src/select/stories/constants.js [hunk 2]
```diff
@@ -65,7 +66,27 @@ const stateOptions = [
   { value: 'WY', label: 'Wyoming' },
 ];
 
+const generateOptions = (count :number) :Object[] => {
```
Comment (UnsungHero97) [inline src/select/stories/constants.js]: Why not just use like `genRandomString()` from `MockUtils`?

File: src/select/stories/constants.js [hunk 3]
```diff
@@ -65,7 +66,27 @@ const stateOptions = [
...
+
+  const options = [];
+  const chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  for (let i = 0; i < count; i += 1) {
+    let label = '';
+    let value = '';
+    for (let j = 0; j < 10; j += 1) {
+      label += chars[Math.floor(Math.random() * chars.length)];
+    }
+    value = chars[Math.floor(Math.random() * chars.length)];
```
Comment (UnsungHero97) [inline src/select/stories/constants.js]: This is always 1 letter. Is this intentional?

File: src/select/stories/constants.js (hunk 2 above)
Comment (UnsungHero97) [reply src/select/stories/constants.js]: Or... use the imported random function above?

Comment (giuliacampana) [reply src/select/stories/constants.js]: 🤦🏽‍♀️🤦🏽‍♀️

File: src/select/stories/constants.js (hunk 3 above)
Comment (giuliacampana) [reply src/select/stories/constants.js]: oops thanks for catching this! it was intentional, but i was just temporarily comparing with have a smaller value vs. a longer one.

**Pull request merged**